SCF/SCF-session1/Assignment-1/ShopingCart.java

```diff
@@ -0,0 +1,213 @@
+	 * @author Ramdev
+	 */
+	class Item
+	{
+		String name;
+		float price;
+		int quantity;
+		public Item(String name,float price,int quantity)
+		{
+			this.name = name;
+			this.price = price;
+			this.quantity = quantity;
+		}
+	}
+	/*
+	 * the size of Item array is fixed at 5 because list of select able items is also 5
+	 */
+	public Item items[] = new Item[5]; 
+	
+	/** Add a item into the cart 
+	 * 
+	 * @param aName
+	 * @param aPrice
+	 * @param aQuantity
+	 * @return Add a item if it is not in the cart, if it is already the in cart then increase the quantity  
+	 */
+	void addItem(String aName, float aPrice, int aQuantity)
+	{
+		int indexOfItem = searchIteam(aName);
+		
+		if(indexOfItem >= 0)
+		{
+			items[indexOfItem].quantity += aQuantity;
+			noOfItems += aQuantity;
+			billOfCart += aPrice*aQuantity;
+		}
+		else
+		{
+			Item item = new Item(aName,aPrice,aQuantity);
+			items[typesOfItems] = item;
+			noOfItems += aQuantity;
+			typesOfItems += 1;
+			billOfCart += aPrice*aQuantity;
+		}
+	}
+	/** to remove any item from cart/ item list 
+	 * 
+	 * @param ID
+	 * @return remove the item from Item list if it is available 
+	 */
+	
+	void removeItem(int ID)
+	{
+		if(ID < 0 || ID > typesOfItems)
+		{
+			System.out.println("wrong Entery found");
+		}
+		else
+		{
+			noOfItems -= items[ID-1].quantity;
+			billOfCart -= (items[ID-1].price)*(items[ID-1].quantity);
+			while( ID < typesOfItems )
+			{
+				items[ID-1] = items[ID];
+				ID += 1;
+			}
+			typesOfItems -= 1;
+		}
+	}
+	/** 
+	 * display the items of cart
+	 */
+	void displayItems()
+	{
+		System.out.println("ID      Name       Price	quantity");
+		for(int iterator = 0; iterator < typesOfItems; iterator++)
+		{
+			System.out.println((iterator + 1) + "		" + items[iterator].name+"	   "+items[iterator].price+"	  "+items[iterator].quantity);
+		}
+	}
+	/**
+	 * 
+	 * @return total bill of cart
+	 */
+	
+	float getBill()
+	{
+		return billOfCart;
+	}
+	/** update the quantity of any item in the cart
+	 * 
+	 * @param uName
+	 * @param uQuantity
+	 */
+	void updateQuantity(String uName, int uQuantity)
+	{
+		if(uQuantity < 0)
+		{
+			System.out.println("Wrong entry found.");
+		}
+		else
+		{
+			int n = searchIteam(uName);
+			if(n != -1)
+			{
+				billOfCart -= billOfCart - items[n].price*items[n].quantity + items[n].price*uQuantity;
+				noOfItems  -= noOfItems  - items[n].quantity + uQuantity;
+				items[n].quantity = uQuantity;
+			}
+			else
+			{
+				System.out.println("item doesnot exists ");
+			}
+		}
+	}
+	/**
+	 * 
+	 * @param sName represents search name
+	 * @return the index of item if it is available in the cart else return -1  
+	 */
+	int searchIteam(String sName)
+	{
+		for(int iterator = 0; iterator < typesOfItems; iterator++)
+		{
+			if(items[iterator].name == sName)
+			{
+				return iterator;
+			}
+		}
+		return -1;
+	}
+	/**
+	 * shows all the choices of items that could be added in to the cart 
+	 */
+	public static void showItems()
```

Comment:
this and displayItems method has same code, we can make a private parameterized method that takes in the list to display and call that from both methods with passing the correct list, or call the parameterized method directly from switch case with proper list passed based on user choice